src/api/app/models/user.rb

```diff
@@ -842,11 +842,11 @@ def to_param
     to_s
   end
 
-  def nr_of_requests_that_need_work
-    Rails.cache.fetch("requests_for_#{login}", expires_in: 2.minutes) do
-      BsRequest.collection(user: login, states: %w(declined), roles: %w(creator)).count +
-      BsRequest.collection(user: login, states: %w(new), roles: %w(maintainer)).count +
-      BsRequest.collection(user: login, roles: %w(reviewer), reviewstates: %w(new), states: %w(review)).count
+  def tasks
+    Rails.cache.fetch("requests_for_#{cache_key}") do
+      declined_requests.count +
+      incoming_requests.count +
+      involved_reviews.count
```

Comment:
I am not aware of a use case where the hiding makes sense, but I have not dig to the git history why the not() is there. Maybe you can have a look.

But I would not hide this request, since you might end up in the situation that the requests waits forever because it does not pop up in the worklist.  It seems the collection code (and therefore also osc) is not hiding this request. Staying in sync with "osc my rq" is also important.